geode-core/src/main/java/org/apache/geode/cache/query/security/RestrictedMethodAuthorizer.java

```diff
@@ -15,147 +15,365 @@
+    acceptanceListMap.put("keySet", regionCallers);
+    acceptanceListMap.put("values", regionCallers);
+    acceptanceListMap.put("getEntries", regionCallers);
+    acceptanceListMap.put("getValues", regionCallers);
+
+    return acceptanceListMap;
+  }
+
+  /**
+   * Returns an unmodifiable view of the methods disallowed by default.
+   * This method can be used to get "read-only" access to the set containing the methods that
+   * are considered non safe by default.
+   *
+   * @return an unmodifiable view of the default disallowed methods.
+   */
+  Set<String> getForbiddenMethods() {
+    return forbiddenMethods;
+  }
+
+  /**
+   * Returns an unmodifiable view of the default allowed methods.
+   * This method can be used to get "read-only" access to the map containing the default set
+   * of allowed methods per class.
+   *
+   * @return an unmodifiable view of the default allowed methods per class map.
+   */
+  Map<String, Set<Class>> getAllowedMethodsPerClass() {
+    return allowedMethodsPerClass;
+  }
+
+  /**
+   * Returns an unmodifiable view of the default Geode allowed methods.
+   * This method can be used to get "read-only" access to the map containing the default set
+   * of allowed geode methods per class.
+   *
+   * @return an unmodifiable view of the default allowed geode methods per class map.
+   */
+  Map<String, Set<Class>> getAllowedGeodeMethodsPerClass() {
+    return allowedGeodeMethodsPerClass;
   }
 
-  protected Map<String, Set> getAcceptList() {
-    return acceptListedMethodsToClass;
+  /**
+   * Creates a {@code RestrictedMethodAuthorizer} object and initializes it so it can be safely
+   * used in a multi-threaded environment.
+   * <p/>
+   *
+   * If the {@link Cache} instance passed as parameter was previously created by Geode, the
+   * authorizer will use the security service already configured in order to determine whether a
+   * specific user has read privileges upon a particular region.
+   * If the {@link Cache} instance passed as parameter is a wrapper created by external frameworks,
+   * the authorizer will create a new instance of the security service using the configuration
+   * properties used to initialize the cache.
+   * <p/>
+   *
+   * Applications can also use this constructor as part of the initialization for custom authorizers
+   * (see {@link Declarable#initialize(Cache, Properties)}), when using a declarative approach.
+   *
+   * @param cache the {@code Cache} instance that owns this authorizer, required in order to
+   *        configure the security rules used.
+   */
+  public RestrictedMethodAuthorizer(Cache cache) {
+    Objects.requireNonNull(cache, "Cache should be provided to configure the authorizer.");
+
+    // Set the correct SecurityService.
+    if (cache instanceof InternalCache) {
+      // Use the already created SecurityService.
+      this.securityService = ((InternalCache) cache).getSecurityService();
+    } else {
```

Comment:
Out of curiosity, when does this branch get invoked?

Reply:
Never in theory, but users or application frameworks might have a wrapper around our cache, I think it's better to have the extra check and create the security service from the properties instead of failing with a `ClassCastException`.